Always coerce non-String attributes upon get() to handle classloader problems with included builds (#32277)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

=== Deprecations

[[null-attribute-lookup]]

> ==== Looking up attributes using null keys deprecated ====

Reviewer: ```suggestion
==== Looking up attributes using null keys is deprecated ====
```

> Providing a `null` argument when retrieving attributes from a container using `getAttribute(Attribute)` is now explicitly deprecated.

Reviewer 2: Can this link to the javadoc?

> Providing a null argument to when retrieving attributes from a container using the `getAttribute(Attribute)` method has been explicitly deprecated.

Reviewer: ```suggestion
Providing a `null` argument when retrieving attributes from a container using `getAttribute(Attribute)` is now explicitly deprecated.
```

> Prior to this change, a `null` lookup silently returned `null`.  Now it will emit a deprecation warning.

Reviewer: ```suggestion
Previously, a `null` lookup silently returned `null`. Now, it emits a deprecation warning.  
```

> There should be no need to lookup null values in an `AttributeContainer`.

Reviewer: ```suggestion
There should be no need to look up `null` values in an `AttributeContainer`.
```
